python/tank/descriptor/descriptor_config.py

```diff
@@ -138,10 +139,16 @@ def _find_interpreter_location(self, path):
                 path_to_python = f.read().strip()
 
             if not path_to_python or not os.path.exists(path_to_python):
-                raise TankInvalidInterpreterLocationError(
-                    "Cannot find interpreter '%s' defined in "
-                    "config file '%s'." % (path_to_python, interpreter_config_file)
+                try:
+                    # Python interpreter could be a bash function
+                    subprocess.check_output("type {}".format(path_to_python), shell=True)
+                except subprocess.CalledProcessError:
```

Comment:
You also need to update the exception that gets thrown here with the one defined inside `tank/util/process.py`